### PR TITLE
replaced defaultdistance direct accesses with weights(g); added const…

### PR DESCRIPTION
**Pull request by sbromberger**
…ructor to DD.

This allows drop-in defaults to work for graph types that already have weights defined.

### CODE REVIEW

**Comment by jpfairbanks** [comment]
This looks good. How do you interpret this with respect to semantic versioning? It changes a behavior of all AbstractGraph types that don't define a weights function does changing default argument types count as changing the API?

**Comment by sbromberger** [comment]
> How do you interpret this with respect to semantic versioning?

This is equivalent code, with an intermediate abstraction, so I'm not worried about it. Plus, http://semver.org/#spec-item-4.

**Comment by sbromberger** [comment]
@jpfairbanks @juliohm if you're ok with this, please approve review? Thanks.

File: src/distance.jl [hunk 1]
```diff
@@ -5,6 +5,11 @@ struct DefaultDistance<:AbstractMatrix{Int}
     DefaultDistance(nv::Int=typemax(Int)) = new(nv)
 end
 
+DefaultDistance(nv::Integer) = DefaultDistance(Int(nv))
```
Comment (juliohm) [inline src/distance.jl]: Could you please elaborate on this line @sbromberger? What integer types are we planning to support?

Comment (sbromberger) [inline src/distance.jl]: The issue is that `nv` now returns `<: Integer`, and I've made `DefaultDistance` just use Ints for simplicity. This converts `nv` to `Int` regardless of the underlying graphtype. It should have no ill effects.

Comment (juliohm) [inline src/distance.jl]: Feel free to merge ;)

Comment (sbromberger) [inline src/distance.jl]: You need to approve via the github interface. :)

**Comment by sbromberger** [comment]
Thanks!

**Pull request merged**